File: src/test/java/br/com/jiratorio/factory/domain/request/HolidayRequestFactory.java
package br.com.jiratorio.factory.domain.request;

import br.com.leonardoferreira.jbacon.JBacon;
import br.com.jiratorio.domain.request.HolidayRequest;
import com.github.javafaker.Faker;
import java.text.SimpleDateFormat;
import java.util.concurrent.TimeUnit;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class HolidayRequestFactory extends JBacon<HolidayRequest> {

    @Autowired
    private Faker faker;

    @Override
    protected HolidayRequest getDefault() {
        HolidayRequest holidayRequest = new HolidayRequest();

        holidayRequest.setDate(new SimpleDateFormat("dd/MM/yyyy")
                .format(faker.date().future(5, TimeUnit.DAYS)));
        holidayRequest.setDescription(faker.lorem().word());

        return holidayRequest;
    }

    @Override
    protected HolidayRequest getEmpty() {
        return new HolidayRequest();
    }

    @Override
    protected void persist(final HolidayRequest holidayRequest) {
        throw new UnsupportedOperationException();
    }
}

File: src/test/java/br/com/jiratorio/factory/domain/request/HolidayRequestFactory.kt
package br.com.jiratorio.factory.domain.request

import br.com.jiratorio.domain.request.HolidayRequest
import br.com.jiratorio.extension.toLocalDate
import br.com.leonardoferreira.jbacon.JBacon
import com.github.javafaker.Faker
import org.springframework.stereotype.Component
import java.util.concurrent.TimeUnit

@Component
class HolidayRequestFactory(
    private val faker: Faker
) : JBacon<HolidayRequest>() {

    override fun getDefault() =
        HolidayRequest(
            date = faker.date().future(5, TimeUnit.DAYS).toLocalDate(),
            description = faker.lorem().word()
        )

    override fun getEmpty() =
        HolidayRequest(
            date = faker.date().future(5, TimeUnit.DAYS).toLocalDate(),
            description = faker.lorem().word()
        )

    override fun persist(holidayRequest: HolidayRequest) =
        throw UnsupportedOperationException()
}
